lib/linter/apply-disable-directives.js

```diff
@@ -261,17 +261,21 @@ function applyDirectives(options) {
     const processed = processUnusedDisableDirectives(unusedDisableDirectivesToReport);
 
     const unusedDisableDirectives = processed
-        .map(({ description, fix, position }) => ({
-            ruleId: null,
-            message: description
-                ? `Unused eslint-disable directive (no problems were reported from ${description}).`
-                : "Unused eslint-disable directive (no problems were reported).",
-            line: position.line,
-            column: position.column,
-            severity: options.reportUnusedDisableDirectives === "warn" ? 1 : 2,
-            nodeType: null,
-            ...options.disableFixes ? {} : { fix }
-        }));
+        .map(({ description, fix, position }) => {
+            const { parentComment, type, line, column } = position;
+
+            return {
+                ruleId: null,
+                message: description
+                    ? `Unused eslint-disable directive (no problems were reported from ${description}).`
+                    : "Unused eslint-disable directive (no problems were reported).",
+                line: type === "disable-next-line" ? parentComment.commentToken.loc.start.line : line,
+                column: type === "disable-next-line" ? parentComment.commentToken.loc.start.column + 1 : column,
```

Comment:
Can we always use `parentComment.commentToken.loc` to simplify?

```suggestion
                line: type === parentComment.commentToken.loc.start.line,
                column: type === parentComment.commentToken.loc.start.column + 1,
```

Reply:
If this would break tests in `tests/lib/linter/apply-disable-directives.js`, then we can leave it as is. In my opinion, those tests require an overly complicated and error-prone setup, and are very difficult to maintain, and it's questionable how useful they are, so I think we could consider dropping them at some point in favor of equivalent tests in `tests/lib/linter/linter.js`.

Reply:
We will have to refactor the `tests/lib/linter/apply-disable-directives.js` to include `loc` - https://github.com/eslint/eslint/blob/eaa08d3055b195bce59cc96bb63ac29038cd7c7d/tests/lib/linter/apply-disable-directives.js#L26-L31

Reply:
That looks too complicated and unmaintainable, so let's leave this as is in the current version of this PR. I think we should (in another PR) entirely delete `tests/lib/linter/apply-disable-directives.js` and convert all those tests to `tests/lib/linter/linter.js` tests.